### PR TITLE
Improve IPv6 parsing

### PR DESCRIPTION
**Pull request by 1st1**
Complimentary to https://github.com/edgedb/shared-client-testcases/pull/11

### CODE REVIEW

File: edgedb/con_utils.py [hunk 1]
```diff
@@ -630,7 +638,8 @@ def _resolve_config_options(
                 resolved_config.set_port(*port)
             if dsn is None:
                 dsn = (
-                    'edgedb://' + (_validate_host(host[0]) if host else ''),
+                    'edgedb://' +
+                    (_prepare_host_for_dsn(host[0]) if host else ''),
```
Comment (elprans) [inline edgedb/con_utils.py:642]: Are we generating a DSN here only to parse it back again?  Is there a better way?

Comment (1st1) [reply edgedb/con_utils.py:644]: @jaclarke

Comment (1st1) [reply edgedb/con_utils.py:644]: Yeah, I'm also wondering why we're doing this that way. Also I'm a bit lost in `host[0]`, `port[0]`, etc

Comment (jaclarke) [reply edgedb/con_utils.py:644]: I think it was so the hostname would be validated the same (by `urllib.parse.urlparse`) whether it was given as `host` or as part of `dsn`

Comment (jaclarke) [reply edgedb/con_utils.py:644]: > Also I'm a bit lost in host[0], port[0], etc

The connection options are passed around as tuples of `(value, source)` to keep track of where they came from for this:

> we'll be including the source of each connection option in the connection error message if one occurs.
https://github.com/edgedb/edgedb/discussions/2922#discussioncomment-1325493

(though it seems that hasn't been fully implemented)

Comment (1st1) [reply edgedb/con_utils.py:644]: @elprans tests pass, I think we should merge this.

**Review by elprans — APPROVED** [review]

**Pull request merged**